test/grpc-fallback.ts

```diff
@@ -0,0 +1,216 @@
+    // tslint:disable-next-line no-any
+    const echoStub: any = await gaxGrpc.createStub(
+      echoService,
+      stubExtraOptions
+    );
+
+    assert(echoStub instanceof protobuf.rpc.Service);
+
+    // The stub should consist of methods
+    expect(echoStub.echo).to.be.a('Function');
+    expect(echoStub.collect).to.be.a('Function');
+    expect(echoStub.chat).to.be.a('Function');
+
+    // There should be 6 methods for the echo service (and 4 other members in the object)
+    assert.strictEqual(Object.keys(echoStub).length, 10);
+
+    // Each of the service methods should take 4 arguments (so that it works with createApiCall)
+    assert.strictEqual(echoStub.echo.length, 4);
+  });
+});
+
+describe('grpc-fallback', () => {
+  let gaxGrpc, protos, echoService, stubOptions;
+  const createdAbortControllers = [];
+  // @ts-ignore
+  const savedAbortController = abortController.AbortController;
+
+  before(() => {
+    stubOptions = {
+      servicePath: 'foo.example.com',
+      port: 443,
+    };
+
+    // @ts-ignore incomplete options
+    gaxGrpc = new GrpcClient(opts);
+    protos = gaxGrpc.loadProto(echoProtoJson);
+    echoService = protos.lookupService('Echo');
+    stubOptions = {
+      servicePath: 'foo.example.com',
+      port: 443,
+    };
+
+    //tslint:disable-next-line variable-name
+    const AbortController = function() {
+      // @ts-ignore
+      this.abort = function() {
+        // @ts-ignore
+        this.abortCalled = true;
+      };
+      // @ts-ignore
+      createdAbortControllers.push(this);
+    };
+
+    // @ts-ignore
+    abortController.AbortController = AbortController;
+  });
+
+  beforeEach(() => {
+    createdAbortControllers.splice(0);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  after(() => {
+    // @ts-ignore
+    abortController.AbortController = savedAbortController;
+  });
+
+  it('should make a request', async () => {
+    const requestObject = {content: 'test-content'};
+    const responseType = protos.lookupType('EchoResponse');
+    const response = responseType.create(requestObject); // request === response for EchoService
+
+    sinon.stub(nodeFetch, 'Promise').returns(
+      Promise.resolve({
+        arrayBuffer: () => {
+          Promise.resolve(responseType.encode(response).finish());
+        },
+      })
+    );
+
+    const echoStub = await gaxGrpc.createStub(echoService, stubOptions);
+    const result = await echoStub.echo(requestObject);
+  });
+
+  it('should be able to cancel an API call using AbortController', async () => {
+    sinon.stub(nodeFetch, 'Promise').returns(Promise.resolve({}));
+
+    const echoStub = await gaxGrpc.createStub(echoService, stubOptions);
+    const request = {content: 'content' + new Date().toString()};
+    const call = echoStub.echo(request, {}, {}, (err, result) => {});
+
+    call.cancel();
+
+    // @ts-ignore
+    assert.strictEqual(createdAbortControllers[0].abortCalled, true);
+  });
```

Comment:
Do we want to add the extra headers test here as well? Since it's in the other grpc-fallback test file as well

Reply:
I tried that but it does not fit well here - mostly because this is supposed to be a unit test and I cannot easily import the `EchoClient` from it (because the `EchoClient` uses `require('google-gax')` and... well, I spent some time trying to make it work and decided we don't want to do it now :)